### PR TITLE
feat: Add local inference option to the API

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Adds a `inference_mode` parameter to the API to make it easier to direct the API to use a local inference server. This is only meant for use when running the running the pipeline API locally because the local model server will not be available on the deployed endpoint. As a future enhancement, we can return an error if local model is used on the deployed endpoint.

### Testing

- Clone the `ml-inference` repo with `git clone https://github.com/Unstructured-IO/ml-inference`.
- Start the `ml-inference` service by running `make run-app-dev` from the `ml-inference directory`.
- Start the OER pipeline API with `UVICORN_PORT=5000 make run-web-app`. The `UVICORN_PORT` variable
  is to deconflict ports with the inference service.
- Make the following API call from the `sample-docs` directory

```
curl -X 'POST' \
  'http://127.0.0.1:5000/oer/v0.0.1/comments' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'files=@fake-oer.pdf;type=application/pdf' \
  -F 'inference_mode=local'
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +84,38 @@ with open(filename, "rb") as f:
...
+
+You can make the same call directly in Python with
+
+```python
+from prepline_oer.api.comments import pipeline_api
+
+filename = "sample-docs/fake-oer.pdf"
+
+with open(filename, "rb") as f:
+    pipeline_api(file=f, filename=filename, m_inference_mode="local")
```
Comment (qued) [inline README.md:114]:
```suggestion
    pipeline_api(file=f, filename=filename, m_inference_mode=["local"])
```
The example as written gets the error `ValueError: Only one value for mode can be passed.`, but with the change it works.

**Review by qued — APPROVED** [review]
Other than the comments, LGTM!

File: README.md (hunk 1 above)
Comment (MthwRobinson) [reply README.md:114]: Oh thanks! I changed that in the code but not in the README

File: README.md [hunk 2]
```diff
@@ -72,8 +72,8 @@ and get the following JSON as the output:
 }
 ```
 
-You can also run the extraction code locally by running the following from the root
-directory of this project:
+You can also run the extraction code with Python directly using the following comments
+root directory of this project:
```
[the remark this answers is not in the document]
Comment (MthwRobinson) [reply README.md]: Updated the wording a bit to make it more explicit

**Pull request merged**